### PR TITLE
Added ansible task to create se

### PR DESCRIPTION
**Pull request by patilshrikant786**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  template:
+    src: user-data.j2
+    dest: /root/{{ kvm_vm_hostname }}/user-data
+
+- name: Copy se qcow2 file
+  command: cp {{ kvm_vm_base_img }} /var/lib/libvirt/images/
+  when: kvm_vm_base_img is defined
+
+- name: Rename se qcow2 file
+  command: mv /var/lib/libvirt/images/{{ kvm_vm_base_img }} {{ IMAGE }}
```
Comment (grastogi23) [inline tasks/main.yml]: Does this conform to the checking of the file. another option to use copy role
https://docs.ansible.com/ansible/latest/modules/copy_module.html

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  when: kvm_vm_base_img is defined
+
+- name: Set the location of cloud image
+  set_fact:
+    USER_DATA: /root/{{ kvm_vm_hostname }}/user-data
```
Comment (grastogi23) [inline tasks/main.yml]: may be have prefix for all the set facts so they don't have collision with other variables.

File: tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  set_fact:
+    META_DATA: /root/{{ kvm_vm_hostname }}/avi_meta-data
+
+- name: Copy se qcow2 file
+  command: cp /root/{{ kvm_vm_hostname }}/se.qcow2 /var/lib/libvirt/images/
+  when: kvm_vm_base_img is undefined
+
+- name: Rename se qcow2 file
+  command: mv /var/lib/libvirt/images/se.qcow2 {{ IMAGE }}
+  when: kvm_vm_base_img is undefined
```
Comment (grastogi23) [inline tasks/main.yml]: same as above. also mv vs copy?

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  register: result
+  with_items: "{{ intf_name['results'] }}"
+
+- name: Initialize an empty string
+  set_fact:
+    host_dv_str: ""
+
+- name: Generate host device string
+  set_fact:
+    host_dv_str:  "{{ host_dv_str }} --host-device={{ item['ansible_facts']['data'] }}"
```
Comment (grastogi23) [inline tasks/main.yml]: you can do this in one shot by providing default. host_dv_str | default("")

File: tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  loop: "{{ result['results'] }}"
+
+- name: Generated host device string
```
Comment (grastogi23) [inline tasks/main.yml]: Let us fix the name so debugging is easy. We should have some convention on how to do it

File: tasks/requirements.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
+---
+- name: check total no of vfs and total intf names are equal or not
+  fail:
+    msg: " Total no of vfs is not equal to the total no of virt_intf_name"
```
Comment (grastogi23) [inline tasks/requirements.yml]: print the values that were passed by the user or that was ultimately resolved for the check.

File: tasks/main.yml (hunk 1 above)
Comment (patilshrikant786) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 2 above)
Comment (patilshrikant786) [reply tasks/main.yml]: Fixed.

File: tasks/main.yml (hunk 4 above)
Comment (patilshrikant786) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 5 above)
Comment (patilshrikant786) [reply tasks/main.yml]: Fixed

File: tasks/requirements.yml (hunk 6 above)
Comment (patilshrikant786) [reply tasks/requirements.yml]: Fixed

File: tasks/main.yml (hunk 3 above)
Comment (patilshrikant786) [reply tasks/main.yml]: Fixed

File: README.md [hunk 7]
```diff
@@ -1,2 +1,72 @@
...
+|kvm_vm_hostname|Yes||Name for VM|
+|kvm_vm_base_img|No||se.qcow2 or controller.qcow2 file|
+|kvm_vm_vcpus|No|4|How many cpus the service engine will use.|
+|kvm_vm_ram|No|8912|How much memory the service engine will use.|
+|kvm_vm_os_disk_size|Yes|40|How much disk size the service engine will use.|
+|host_mgmt_intf|Yes||host management interface name|
+|vm_ctrl_ip|Yes||The IP address of the controller.|
+|vm_ctrl_username|Yes||The username to login into the controller.|
+|vm_ctrl_password|Yes||The password to login into the controller.|
+|vm_ctrl_version|Yes||The controller version.|
```
Comment (MUKULSINHA) [inline README.md]: Was this an input being taken in CSP ? Have we borrowed this variable from there ?

**Pull request merged**